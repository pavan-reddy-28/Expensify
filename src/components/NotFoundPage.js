import React from 'react'
import {Link} from 'react-router-dom'
const NotFoundPage = () => (
    <>
        <h3>
        404  <Link to="/">Go To DashBoard</Link>
        </h3>
    </>
)
export default NotFoundPage
