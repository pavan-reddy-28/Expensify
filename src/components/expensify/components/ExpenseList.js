import React from 'react'
import {connect} from 'react-redux'
import ExpenseListItem from './ExpenseListItem'

import expensesSelectors from '../redux/selectors/expenses'

export const ExpenseList=(props)=>{
    console.log(props.expenses)
    return (
        <div>
            
            {
                props.expenses.length===0?
                <h2>No Expenses</h2>:
                <>
                <h2>ExpenseList</h2>
                    {props.expenses.map(expense=>(
                    <ExpenseListItem
                        key={expense.id}
                        {...expense}   
                    />
                    ))}
                </>
            }
                
        </div>
    )
}
const mapStateToProps  =(state)=>{
    console.log(state.filters)
    return{
        expenses:expensesSelectors(state.expenses,state.filters)
    }
}

export default connect(mapStateToProps)(ExpenseList);
