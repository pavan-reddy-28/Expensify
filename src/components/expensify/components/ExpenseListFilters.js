import React from 'react'
import {connect} from 'react-redux'
import {setTextFilter,sortByAmount,sortByDate} from '../reduxActionExports'
function ExpenseFilterInput(props) {
    return (
        <div style={{paddind:'10px',margin:'5px'}}>
            <input type='text' value={props.filters.text} onChange={(e)=>{
                console.log(e.target.value)
                props.dispatch(setTextFilter(e.target.value))
            }}/>

            Sort by:
            <select

                value={props.filters.sortBy}
                onChange={(e)=>{
                   
                if(e.target.value==='amount'){
                    props.dispatch(sortByAmount('amount'))
                }else{
                    props.dispatch(sortByDate('date'))
                }
            }} style={{paddind:'10px',margin:'5px'}}>
                <option value="time">Created Time</option>
                <option value="amount">Amount</option>
            </select>
       
        </div>
    )
}

const mapStateToProps = (state)=>{
    return {filters:state.filters}
}
export default connect(mapStateToProps)(ExpenseFilterInput)