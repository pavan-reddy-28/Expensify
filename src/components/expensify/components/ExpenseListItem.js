import React from 'react'
import {connect} from 'react-redux'
import {removeExpense} from '../reduxActionExports'
import {Link} from 'react-router-dom'
export const  ExpenseListItem=({id,description,amount,createdAt})=>{
    return (
        <div>
            <Link to={`/edit/${id}`}>            
                <h2>{description}</h2>
            </Link>

    <div>
       <p>Amount : <b>{amount}</b>  {'   '}  Created Time :<b>{createdAt}</b></p>
      
       {/* <button onClick={()=>dispatch(removeExpense(id))}>Remove</button> */}
    </div>
        </div>
    )
}

export default connect()(ExpenseListItem)
