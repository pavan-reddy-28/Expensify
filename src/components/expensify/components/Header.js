import React from 'react'
import {NavLink} from 'react-router-dom'
const Header = () => (
    <div >
        <h1>
            Expensify
</h1>
        <NavLink className="nav-link-el" to="/" activeClassName="is-active" exact> DashBoard</NavLink>
       
        <NavLink className="nav-link-el" to="/create" activeClassName="is-active">Create</NavLink>
        <NavLink className="nav-link-el" to="/help" activeClassName="is-active">Help</NavLink>

    </div>
)


export default Header
