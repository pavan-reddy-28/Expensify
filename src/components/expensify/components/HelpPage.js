import React from 'react'

const HelpPage = () => (
    <>
        <h3>
            help route
    </h3>

    </>
)

export default HelpPage
