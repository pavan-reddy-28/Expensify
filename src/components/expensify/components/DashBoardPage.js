import React from 'react'
import ExpenseList from './ExpenseList'

import ExpenseFilterInput from './ExpenseListFilters'
function DashBoardPage() {
    return (<>
        <ExpenseFilterInput/>
       <ExpenseList/>
       </>
    )
}

export default DashBoardPage
