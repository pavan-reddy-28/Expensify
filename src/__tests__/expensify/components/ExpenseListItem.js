import React from 'react'
import {shallow} from 'enzyme'
import {ExpenseListItem } from '../../../components/expensify/components/ExpenseListItem'
import ExpenseArray from '../../../fakeExpensesArray/fakeData'

test('should render expense list item with expenseArray data ', () => {
    const wrapper = shallow(<ExpenseListItem {...ExpenseArray[0]} />)
    expect(wrapper).toMatchSnapshot()
})
