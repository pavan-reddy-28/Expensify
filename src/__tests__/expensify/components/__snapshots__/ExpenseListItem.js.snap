// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense list item with expenseArray data  1`] = `
<div>
  <Link
    to="/edit/0"
  >
    <h2>
      gas bill
    </h2>
  </Link>
  <div>
    <p>
      Amount : 
      <b>
        1234
      </b>
        
         
        Created Time :
      <b>
        0
      </b>
    </p>
  </div>
</div>
`;
